config/routes.rb

```diff
@@ -1,5 +1,17 @@
 Rails.application.routes.draw do
-  root "pokemon#index"
+  get 'favorites/create'
+  get 'favorites/index'
+  root "search#pokelist"
 
-  get "/pokemon", to: "pokemon#index"
+  scope '/v1' do
+    get "/pokemon", to: "pokemon#index"
+    get "/search", to: "search#pokelist"
+    get "/", to: "search#pokelist"
+  end
+  scope '/v2' do
+    get "/", to: "favorites#all"
+  end
+  resources :pokemon, only: [:index, :show]
+  resources :favorites, only: [:create, :index]
```

Comment:
```suggestion
  resources :favorites, only: [:create, :index, :destroy]
```

Colocando assim, você poderia tirar o delete 'favorites/clear', to: 'favorites#clear', as: 'clear_favorites', e depois renomear o método de clear para destroy

Reply:
Posso por, mas em específico o clear deleta todos os favoritos do usuário e não só um em especifico, é basicamente uma limpa